File: algorithm/tree/balanced_binary_tree/c/src/node.c
#include "node.h"
#include <stddef.h>

int get_height(struct Node *node) {
    if (node == NULL) {
        return -1;
    }
    return node->height;
}

int get_factor(struct Node *node) {
    if (node == NULL) {
        return 0;
    }
    return get_height(node->left) - get_height(node->right);
}

File: algorithm/tree/balanced_binary_tree/java/Node.java
public class Node {
    public int val;
    public int height;
    public Node parent;
    public Node left;
    public Node right;

    public Node() {
    }
    public Node(int val) {
        this.val = val;
        this.height = 0;
        this.parent = null;
        this.left = null;
        this.right = null;
    }
    public static int getHeight(Node node) {
        if (node == null) {
            return -1;
        }
        return node.height;
    }
    public static int getFactor(Node node) {
        return Node.getHeight(node.left) - Node.getHeight(node.right);
    }
}
